src/annotator/util/clean-context.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * Export clean versions of standard functions and objects.
+ *
+ * This module exports "clean" versions of various standard functions that may
+ * have been modified in the parent browsing context that the client is
+ * injected into. Code should always use the clean versions from this module,
+ * and never the potentially modified versions from the parent context.
+ *
+ * @module util/clean-context
+ */
+
+'use strict';
+
+// Append an invisible <iframe> to the parent document in order to get a clean
+// browsing context to extract clean objects and functions from.
+var iframe = document.createElement('iframe');
+iframe.style.display = 'none';
+document.body.appendChild(iframe);
```

Comment:
Or of someone just modifies `Function.prototype.bind.something` but doesn't change `toString`?

Let's see if we run into anyone that does it, I guess. I think this clean context thing is a temporary batch (before moving all of `src/annotator` into a same-origin iframe) anyway.

If we want to move the whole thing into an iframe by the way, when we do that we will have to eat this 6-100ms delay every time anyway won't we?